bokeh/application/handlers/directory.py

```diff
@@ -32,6 +36,10 @@ def url_path(self):
     def modify_document(self, doc):
         if self.failed:
             return
+        # Note: we do NOT copy self._theme, which assumes the Theme
+        # class is immutable (has no setters)
```

Comment:
off the cuff: a class decorator if `Theme` does not end up being a `Model`, or some "read only policy` for `MetaHasProps` if it does.